### PR TITLE
Replace resources api to return the max allocatable memory

### PR DESCRIPTION
**Pull request by cesnietor**
Resouces api changed to GET on `/api/v1/cluster/max-allocatable-memory`
which returns the max allocatable memory for a given number of nodes, 
nodes with taints NoExecute or NoSchedule shall not be considered.
request needs to be e.g.
```
/api/v1/cluster/max-allocatable-memory?num_nodes=4
```
Response payload:
```
{
    "max_memory": 4131528704
}
```
If not enough nodes or no available memory response will be empty `{}`

- [X] Tests included

### CODE REVIEW

File: swagger.yml [hunk 1]
```diff
@@ -1227,15 +1227,21 @@ paths:
       tags:
         - AdminAPI
   
-  /cluster/resources:
```
Comment (bexsoft) [inline swagger.yml:1230]: Is this is no longer needed?

Comment (cesnietor) [reply swagger.yml:1230]: not anymore, this was meant to be for the same feature of calculating the max ram but we'll do it in this new api

**Review by Alevsk — APPROVED** [review]

File: swagger.yml (hunk 1 above)
Comment (dvaldivia) [reply swagger.yml:1230]: I think resources could also express storage capacity if available right? I think nesting this API under resources made sense

Comment (cesnietor) [reply swagger.yml:1230]: that api is no longer needed, was designed for the main purpose

**Comment by cesnietor** [comment]
PR modified to use queryparameter instead of body parameter

**Review by dvaldivia — APPROVED** [review]

**Review by Alevsk — APPROVED** [review]

**Pull request merged**